### PR TITLE
Type-safe CSS Stitches' variables

### PR DESCRIPTION
**Pull request by rpominov**
## Description

I've replaced all `"$something$something"` CSS variables with `variables.something.something`. Now we have autocomplete and type-safety: if a variable is removed or renamed w/o fixing the usage the build will fail. 

This will allow designers to change design tokens and either make sure nothing is broken or if something is broken we'll know exactly what and it will be easy to fix.

Discussed here: https://discord.com/channels/955905230107738152/1061992022665334837

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/.storybook/preview.tsx
Comment (kof) [inline apps/designer/.storybook/preview.tsx:4]: variables in js context is a bit unclear, what about themeVars?  or stitchesVars? since its a particular format for the value defined by stitches?

**Review by kof — APPROVED** [review]

File: apps/designer/.storybook/preview.tsx
Comment (rpominov) [reply apps/designer/.storybook/preview.tsx]: Talked in dm. Decided to use `theme`. But need to rename an existing `theme` export in `stitches.config.ts`. Will rename in separate PR.

**Comment by rpominov** [comment]
Other than `variables` -> `theme` this PR is ready, just will need to apply codemod right before the merge

**Comment by kof** [comment]
Already approved, feel free to merge once renamed

File: apps/designer/app/designer/features/breakpoints/breakpoints.tsx
Comment (TrySound) [inline apps/designer/app/designer/features/breakpoints/breakpoints.tsx:123]: Is this indexed access safe? Can you use here out of bound index?

Comment (rpominov) [reply apps/designer/app/designer/features/breakpoints/breakpoints.tsx:123]: Good point! But yes, seems like safe https://www.typescriptlang.org/play?ts=4.9.4#code/MYewdgzgLgBFCm0YF4YG8BQNswEQEZcAuGfAGix1wCZiZqKBfGAQwhlEigG4MNOk4eCjiIoAbXwBdfuCRQAFgCd4w1AmjiAzFKA

**Pull request merged**
